### PR TITLE
Adding implementation to publish the data of a study to Kafka after it's been registered

### PR DESCRIPTION
**Pull request by pgadzios**
Adding a message bus producer to publish serialized events of the received data to be available for the Data Platform to consume, via NSIP.
There are some existing projects available in the NIHR.NSIP solution space which should be leveraged in future (i.e. to re-use code) however as they're not packaged as nuget packages currently, we just wrote a basic Kafka implementation for the PoC. Beyond PoC we should consider publishing those assemblies as nuget packages as an opportunity for code re-use.

### CODE REVIEW

**Comment by pgadzios** [comment]
Hi @con-nihr are you able to please cast your eye over this PR when you have an opportunity.

File: NIHR.StudyManagement/NIHR.StudyManagement.Domain/Services/GovernmentResearchIdentifierService.cs [hunk 1]
```diff
@@ -72,7 +75,11 @@ public class GovernmentResearchIdentifierService : IGovernmentResearchIdentifier
                 ProtocolId = request.ProtocolId
             };
 
-            return await _governmentResearchIdentifierRepository.AddStudyToIdentifierAsync(domainRequest, cancellationToken);
+            var researchStudy = await _governmentResearchIdentifierRepository.AddStudyToIdentifierAsync(domainRequest, cancellationToken);
+
+            await _messagePublisher.PublishAsync("gris-study-registered", _settings.DefaultLocalSystemName, researchStudy, cancellationToken);
```
Comment (con-nihr) [inline NIHR.StudyManagement/NIHR.StudyManagement.Domain/Services/GovernmentResearchIdentifierService.cs:80]: Can you pull out the event type strings into a central 'registry' of `consts`?

**Review by con-nihr — CHANGES_REQUESTED** [review]
Agreed, we need to making some reusable NSIP abstractions. That's a job for later.

File: NIHR.StudyManagement/NIHR.StudyManagement.Infrastructure/MessageBus/StudyManagementKafkaMessageProducer.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        "Status={Status} " +
+                        "Value={Value}",
+                        _settings.Topic,
+                        deliveryReport.Offset.Value,
+                        "PoC" ?? "NULL",
+                        deliveryReport.TopicPartition.Partition.Value,
+                        deliveryReport.Status,
+                        nsipMessageJson);
+                }
+                catch (Exception ex)
```
[the remark this answers is not in the document]
Comment (pgadzios) [reply NIHR.StudyManagement/NIHR.StudyManagement.Infrastructure/MessageBus/StudyManagementKafkaMessageProducer.cs]: Yep, classic example of copy-paste :) from this file https://github.com/PA-NIHR-CRN/nihr-nsip/blob/main/NIHR.NSIP/NIHR.NSIP.KafkaSdk/KafkaProducer.cs

This try/catch has now been removed.

File: NIHR.StudyManagement/NIHR.StudyManagement.Domain/Services/GovernmentResearchIdentifierService.cs (hunk 1 above)
Comment (pgadzios) [reply NIHR.StudyManagement/NIHR.StudyManagement.Domain/Services/GovernmentResearchIdentifierService.cs]: Magic string removed into constant GrisNsipEventTypes

**Pull request merged**